src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,201 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteTaskCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.SortByDateCommand;
+import duke.commands.SortByNameCommand;
+import duke.exceptions.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+public class Parser {
+
+    public static Task parseToTask(String userInput) throws DukeException {
+
+        if (userInput.startsWith("todo")) {
+            return Parser.newToDo(userInput);
+        } else if (userInput.startsWith("deadline")) {
+            return Parser.newDeadline(userInput);
+        } else if (userInput.startsWith("event")) {
+            return Parser.newEvent(userInput);
+        } else {
+            // Not used
+            return null;
+        }
+    }
+    private static Todo newToDo(String s) throws DukeException {
```

Comment:
I think it would be more consistent to have these newX() method names to be phrased as verbs.